examples/keyboard/Keyboard.cpp

```diff
@@ -0,0 +1,864 @@
+        const float alpha = std::max(0.f, ratio * (2.f - ratio)) * 0.5f;
+
+        sf::Color color = getOutlineColor();
+        color.a         = static_cast<std::uint8_t>(255 * alpha);
+        setOutlineColor(color);
+
+        if (m_remaining > sf::Time::Zero)
+            m_remaining -= frameTime;
+    }
+
+private:
+    static constexpr sf::Time duration = sf::milliseconds(150);
+    sf::Time                  m_remaining;
+};
+
+////////////////////////////////////////////////////////////
+// Utilities to create text objets
+////////////////////////////////////////////////////////////
+
+constexpr unsigned int textSize = 18u;
+constexpr unsigned int space    = 2u;
+constexpr unsigned int lineSize = textSize + space;
+
+float getSpacingFactor(const sf::Font& font)
+{
+    return static_cast<float>(lineSize) / font.getLineSpacing(textSize);
+}
+
+ShinyText makeShinyText(const sf::Font& font, const sf::String& string, const sf::Vector2f& position)
+{
+    ShinyText text(font, string, textSize);
+    text.setLineSpacing(getSpacingFactor(font));
+    text.setOutlineThickness(2.f);
+    text.setPosition(position);
+
+    return text;
+}
+
+sf::Text makeText(const sf::Font& font, const sf::String& string, const sf::Vector2f& position)
+{
+    sf::Text text(font, string, textSize);
+    text.setLineSpacing(getSpacingFactor(font));
+    text.setPosition(position);
+
+    return text;
+}
+
+////////////////////////////////////////////////////////////
+// Utilities to describe keyboard events
+////////////////////////////////////////////////////////////
+
+template <typename KeyEventType>
+bool somethingIsOdd(const KeyEventType& keyEvent)
+{
+    return keyEvent.code == sf::Keyboard::Key::Unknown || keyEvent.scancode == sf::Keyboard::Scan::Unknown ||
+           sf::Keyboard::getDescription(keyEvent.scancode) == "" ||
+           sf::Keyboard::localize(keyEvent.scancode) != keyEvent.code ||
+           sf::Keyboard::delocalize(keyEvent.code) != keyEvent.scancode;
+}
+
+// Append information to string about a keyboard event
+template <typename KeyEventType>
+sf::String keyEventDescription(sf::String text, const KeyEventType& keyEvent)
+{
+    text += "\n\n";
+    text += keyIdentifier(keyEvent.code);
+    text += "\n";
+    text += scancodeIdentifier(keyEvent.scancode);
+    if (somethingIsOdd(keyEvent))
+    {
+        text += "\nLocalized:\t";
+        text += keyIdentifier(sf::Keyboard::localize(keyEvent.scancode));
+        text += "\nDelocalized:\t";
+        text += scancodeIdentifier(sf::Keyboard::delocalize(keyEvent.code));
+    }
+
+    return text;
+}
+
+// Make a string describing a text event
+sf::String textEventDescription(const sf::Event::TextEntered& textEntered)
+{
+    sf::String text = "Text Entered\n\n";
+    text += static_cast<char32_t>(textEntered.unicode);
+    text += "\nU+";
+
+    std::ostringstream oss;
+    oss << std::hex << std::setw(4) << std::setfill('0') << textEntered.unicode;
+
+    return text += oss.str();
```

Comment:
@ChrisThrasher I will revert to retuning `text` on its own to enable implicit move instead of copy.